pandas/core/indexes/datetimelike.py

```diff
@@ -191,10 +191,8 @@ def sort_values(self, return_indexer=False, ascending=True):
             sorted_index = self.take(_as)
             return sorted_index, _as
         else:
-            # NB: using asi8 instead of _ndarray_values matters in numpy 1.18
-            #  because the treatment of NaT has been changed to put NaT last
-            #  instead of first.
-            sorted_values = np.sort(self.asi8)
+            values = self._data
+            sorted_values = values[values.argsort()].asi8
```

Comment:
I don't think we want this... This is ~2x slower than passing data to `np.sort`.

For datetime / timedelta, I think we're OK passing `.view(M8[ns])` and `.view(m8[ns])` to `np.sort`.

Looking into alternatives for period.